glean-core/rlb/src/system.rs

```diff
@@ -53,3 +53,37 @@ pub const ARCH: &str = "x86_64";
 )))]
 /// `target_arch` when building this crate: unknown!
 pub const ARCH: &str = "unknown";
+
+#[cfg(any(target_os = "macos", target_os = "windows"))]
+/// Returns Darwin kernel version for MacOS, or NT Kernel version for Windows
+pub fn get_os_version() -> String {
+    sys_info::os_release().unwrap_or_else(|_| "unknown".to_owned())
+}
+
+#[cfg(not(any(target_os = "linux", target_os = "macos", target_os = "windows")))]
+/// Returns "unknown" for platforms other than Linux, MacOS or Windows
+pub fn get_os_version() -> String {
+    "unknown".to_owned()
+}
+
+#[cfg(target_os = "linux")]
+/// Returns Linux kernel version, in the format of <Major>.<Minor> e.g. 5.8
+pub fn get_os_version() -> String {
+    parse_linux_os_string(sys_info::os_release().unwrap_or_else(|_| "unknown".to_owned()))
+}
+
+#[cfg(target_os = "linux")]
+fn parse_linux_os_string(os_str: String) -> String {
+    os_str.split('.').take(2).collect::<Vec<&str>>().join(".")
+}
+
+#[test]
+#[cfg(target_os = "linux")]
+fn parse_fixed_linux_os_string() {
+    let alpine_os_string = "4.12.0-rc6-g48ec1f0-dirty".to_owned();
+    assert_eq!(parse_linux_os_string(alpine_os_string), "4.12".to_owned());
```

Comment:
Tip: in theory you should be able to compare a `&str` with `String` here. Might need to swap things around though, but it's also not super important for testing.

```suggestion
    assert_eq!("4.12", parse_linux_os_string(alpine_os_string));
```